--- tools/build/src_ori/engine/builtins.cpp	2026-05-05 16:01:50.878134393 +0000
+++ tools/build/src/engine/builtins.cpp	2026-05-05 16:04:01.343811314 +0000
@@ -23,7 +23,7 @@
 #include "parse.h"
 #include "pathsys.h"
 #include "rules.h"
-#include "strings.h"
+#include "jam_strings.h"
 #include "subst.h"
 #include "timestamp.h"
 #include "variable.h"
--- tools/build/src_ori/engine/class.cpp	2026-05-05 16:01:50.878134393 +0000
+++ tools/build/src/engine/class.cpp	2026-05-05 16:04:01.344811319 +0000
@@ -13,7 +13,7 @@
 #include "lists.h"
 #include "object.h"
 #include "rules.h"
-#include "strings.h"
+#include "jam_strings.h"
 #include "variable.h"
 #include "output.h"
 
--- tools/build/src_ori/engine/command.h	2026-05-05 16:01:50.878134393 +0000
+++ tools/build/src/engine/command.h	2026-05-05 16:04:01.343811314 +0000
@@ -43,7 +43,7 @@
 #include "config.h"
 #include "lists.h"
 #include "rules.h"
-#include "strings.h"
+#include "jam_strings.h"
 
 
 typedef struct _cmd CMD;
--- tools/build/src_ori/engine/compile.cpp	2026-05-05 16:01:50.879134398 +0000
+++ tools/build/src/engine/compile.cpp	2026-05-05 16:04:01.342811308 +0000
@@ -33,7 +33,7 @@
 #include "parse.h"
 #include "rules.h"
 #include "search.h"
-#include "strings.h"
+#include "jam_strings.h"
 #include "variable.h"
 #include "output.h"
 
--- tools/build/src_ori/engine/debugger.cpp	2026-05-05 16:01:50.879134398 +0000
+++ tools/build/src/engine/debugger.cpp	2026-05-05 16:04:01.345811324 +0000
@@ -7,7 +7,7 @@
 
 #include "debugger.h"
 #include "constants.h"
-#include "strings.h"
+#include "jam_strings.h"
 #include "pathsys.h"
 #include "cwd.h"
 #include "function.h"
--- tools/build/src_ori/engine/execcmd.h	2026-05-05 16:01:50.877134388 +0000
+++ tools/build/src/engine/execcmd.h	2026-05-05 16:04:01.344811319 +0000
@@ -17,7 +17,7 @@
 
 #include "config.h"
 #include "lists.h"
-#include "strings.h"
+#include "jam_strings.h"
 #include "timestamp.h"
 
 
--- tools/build/src_ori/engine/execunix.cpp	2026-05-05 16:01:50.878134393 +0000
+++ tools/build/src/engine/execunix.cpp	2026-05-05 16:04:01.340811298 +0000
@@ -10,7 +10,7 @@
 
 #include "lists.h"
 #include "output.h"
-#include "strings.h"
+#include "jam_strings.h"
 
 #include <errno.h>
 #include <signal.h>
--- tools/build/src_ori/engine/filent.cpp	2026-05-05 16:01:50.878134393 +0000
+++ tools/build/src/engine/filent.cpp	2026-05-05 16:04:01.335811273 +0000
@@ -34,7 +34,7 @@
 
 #include "object.h"
 #include "pathsys.h"
-#include "strings.h"
+#include "jam_strings.h"
 #include "output.h"
 
 #ifdef __BORLANDC__
--- tools/build/src_ori/engine/filesys.cpp	2026-05-05 16:01:50.879134398 +0000
+++ tools/build/src/engine/filesys.cpp	2026-05-05 16:04:01.342811308 +0000
@@ -33,7 +33,7 @@
 #include "lists.h"
 #include "object.h"
 #include "pathsys.h"
-#include "strings.h"
+#include "jam_strings.h"
 #include "output.h"
 
 #include <assert.h>
--- tools/build/src_ori/engine/fileunix.cpp	2026-05-05 16:01:50.878134393 +0000
+++ tools/build/src/engine/fileunix.cpp	2026-05-05 16:04:01.344811319 +0000
@@ -33,7 +33,7 @@
 
 #include "object.h"
 #include "pathsys.h"
-#include "strings.h"
+#include "jam_strings.h"
 #include "output.h"
 
 #include <assert.h>
--- tools/build/src_ori/engine/filevms.cpp	2026-05-05 16:01:50.879134398 +0000
+++ tools/build/src/engine/filevms.cpp	2026-05-05 16:04:01.337811283 +0000
@@ -18,7 +18,7 @@
 
 #include "object.h"
 #include "pathsys.h"
-#include "strings.h"
+#include "jam_strings.h"
 
 
 #ifdef OS_VMS
--- tools/build/src_ori/engine/function.h	2026-05-05 16:01:50.879134398 +0000
+++ tools/build/src/engine/function.h	2026-05-05 16:04:01.342811308 +0000
@@ -12,7 +12,7 @@
 #include "frames.h"
 #include "lists.h"
 #include "parse.h"
-#include "strings.h"
+#include "jam_strings.h"
 
 typedef struct _function FUNCTION;
 typedef struct _stack STACK;
--- tools/build/src_ori/engine/hdrmacro.cpp	2026-05-05 16:01:50.879134398 +0000
+++ tools/build/src/engine/hdrmacro.cpp	2026-05-05 16:04:01.339811293 +0000
@@ -36,7 +36,7 @@
 #include "object.h"
 #include "parse.h"
 #include "rules.h"
-#include "strings.h"
+#include "jam_strings.h"
 #include "subst.h"
 #include "variable.h"
 #include "output.h"
--- tools/build/src_ori/engine/jam.cpp	2026-05-05 16:01:50.879134398 +0000
+++ tools/build/src/engine/jam.cpp	2026-05-05 16:04:01.338811288 +0000
@@ -130,7 +130,7 @@
 #include "rules.h"
 #include "scan.h"
 #include "search.h"
-#include "strings.h"
+#include "jam_strings.h"
 #include "timestamp.h"
 #include "variable.h"
 #include "execcmd.h"
--- tools/build/src_ori/engine/jam_strings.cpp	1970-01-01 00:00:00.000000000 +0000
+++ tools/build/src/engine/jam_strings.cpp	2026-05-05 16:04:01.344811319 +0000
@@ -0,0 +1,252 @@
+/* Copyright David Abrahams 2004. Distributed under the Boost */
+/* Software License, Version 1.0. (See accompanying */
+/* file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt) */
+
+#include "jam.h"
+#include "jam_strings.h"
+
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+
+#ifndef NDEBUG
+# define JAM_STRING_MAGIC ((char)0xcf)
+# define JAM_STRING_MAGIC_SIZE 4
+static void assert_invariants( string * self )
+{
+    int i;
+
+    if ( self->value == 0 )
+    {
+        assert( self->size == 0 );
+        assert( self->capacity == 0 );
+        assert( self->opt[ 0 ] == 0 );
+        return;
+    }
+
+    assert( self->size < self->capacity );
+    assert( ( self->capacity <= sizeof( self->opt ) ) == ( self->value == self->opt ) );
+    assert( self->value[ self->size ] == 0 );
+    /* String objects modified manually after construction to contain embedded
+     * '\0' characters are considered structurally valid.
+     */
+    assert( strlen( self->value ) <= self->size );
+
+    for ( i = 0; i < 4; ++i )
+    {
+        assert( self->magic[ i ] == JAM_STRING_MAGIC );
+        assert( self->value[ self->capacity + i ] == JAM_STRING_MAGIC );
+    }
+}
+#else
+# define JAM_STRING_MAGIC_SIZE 0
+# define assert_invariants(x) do {} while (0)
+#endif
+
+
+void string_new( string * s )
+{
+    s->value = s->opt;
+    s->size = 0;
+    s->capacity = sizeof( s->opt );
+    s->opt[ 0 ] = 0;
+#ifndef NDEBUG
+    memset( s->magic, JAM_STRING_MAGIC, sizeof( s->magic ) );
+#endif
+    assert_invariants( s );
+}
+
+
+void string_free( string * s )
+{
+    assert_invariants( s );
+    if ( s->value != s->opt )
+        BJAM_FREE( s->value );
+    string_new( s );
+}
+
+
+static void string_reserve_internal( string * self, size_t capacity )
+{
+    if ( self->value == self->opt )
+    {
+        self->value = (char *)BJAM_MALLOC_ATOMIC( capacity +
+            JAM_STRING_MAGIC_SIZE );
+        self->value[ 0 ] = 0;
+        strncat( self->value, self->opt, sizeof(self->opt) );
+        assert( strlen( self->value ) <= self->capacity && "Regression test" );
+    }
+    else
+    {
+        self->value = (char *)BJAM_REALLOC( self->value, capacity +
+            JAM_STRING_MAGIC_SIZE );
+    }
+#ifndef NDEBUG
+    memcpy( self->value + capacity, self->magic, JAM_STRING_MAGIC_SIZE );
+#endif
+    self->capacity = capacity;
+}
+
+
+void string_reserve( string * self, size_t capacity )
+{
+    assert_invariants( self );
+    if ( capacity <= self->capacity )
+        return;
+    string_reserve_internal( self, capacity );
+    assert_invariants( self );
+}
+
+
+static void extend_full( string * self, char const * start, char const * finish )
+{
+    size_t new_size = self->capacity + ( finish - start );
+    size_t new_capacity = self->capacity;
+    size_t old_size = self->capacity;
+    while ( new_capacity < new_size + 1)
+        new_capacity <<= 1;
+    string_reserve_internal( self, new_capacity );
+    memcpy( self->value + old_size, start, new_size - old_size );
+    self->value[ new_size ] = 0;
+    self->size = new_size;
+}
+
+static void maybe_reserve( string * self, size_t new_size )
+{
+    size_t capacity = self->capacity;
+    if ( capacity <= new_size )
+    {
+        size_t new_capacity = capacity;
+        while ( new_capacity <= new_size )
+            new_capacity <<= 1;
+        string_reserve_internal( self, new_capacity );
+    }
+}
+
+
+void string_append( string * self, char const * rhs )
+{
+    size_t rhs_size = strlen( rhs );
+    size_t new_size = self->size + rhs_size;
+    assert_invariants( self );
+
+    maybe_reserve( self, new_size );
+
+    memcpy( self->value + self->size, rhs, rhs_size + 1 );
+    self->size = new_size;
+
+    assert_invariants( self );
+}
+
+
+void string_append_range( string * self, char const * start, char const * finish )
+{
+    size_t rhs_size = finish - start;
+    size_t new_size = self->size + rhs_size;
+    assert_invariants( self );
+
+    maybe_reserve( self, new_size );
+
+    if ( start != finish )
+        memcpy( self->value + self->size, start, rhs_size );
+    self->size = new_size;
+    self->value[ new_size ] = 0;
+
+    assert_invariants( self );
+}
+
+
+void string_copy( string * s, char const * rhs )
+{
+    string_new( s );
+    string_append( s, rhs );
+}
+
+void string_truncate( string * self, size_t n )
+{
+    assert_invariants( self );
+    assert( n <= self->capacity );
+    self->value[ self->size = n ] = 0;
+    assert_invariants( self );
+}
+
+
+void string_pop_back( string * self )
+{
+    string_truncate( self, self->size - 1 );
+}
+
+
+void string_push_back( string * self, char x )
+{
+    string_append_range( self, &x, &x + 1 );
+}
+
+
+char string_back( string * self )
+{
+    assert_invariants( self );
+    return self->value[ self->size - 1 ];
+}
+
+void string_rtrim( string * self )
+{
+    char *p;
+    assert_invariants( self );
+    p = self->value + self->size - 1;
+    for ( ; p >= self->value && ( *p == '\0' || isspace( *p ) ); *p-- = 0 );
+}
+
+#ifndef NDEBUG
+void string_unit_test()
+{
+    {
+        string s[ 1 ];
+        int i;
+        int const limit = sizeof( s->opt ) * 2 + 2;
+        string_new( s );
+        assert( s->value == s->opt );
+        for ( i = 0; i < limit; ++i )
+        {
+            string_push_back( s, (char)( i + 1 ) );
+            assert( s->size == i + 1 );
+        }
+        assert( s->size == limit );
+        assert( s->value != s->opt );
+        for ( i = 0; i < limit; ++i )
+            assert( s->value[ i ] == (char)( i + 1 ) );
+        string_free( s );
+    }
+
+    {
+        const char * const original = "  \n\t\v  Foo \r\n\v \tBar\n\n\r\r\t\n\v\t \t";
+        string copy[ 1 ];
+        string_copy( copy, original );
+        assert( !strcmp( copy->value, original ) );
+        assert( copy->size == strlen( original ) );
+        string_free( copy );
+    }
+
+    {
+        const char * const foo = "Foo    ";
+        string foo_copy[ 1 ];
+        string_copy( foo_copy, foo );
+        string_rtrim( foo_copy );
+        assert( !strcmp( foo_copy->value, "Foo" ) );
+
+        string_rtrim( foo_copy );
+        assert( !strcmp( foo_copy->value, "Foo" ) );
+    }
+    {
+        const char * const bar = "Bar\0\0\0";
+        string bar_copy[ 1 ];
+        string_copy( bar_copy, bar );
+        string_rtrim( bar_copy );
+        assert( !strcmp( bar_copy->value, "Bar" ) );
+
+        string_rtrim( bar_copy );
+        assert( !strcmp( bar_copy->value, "Bar" ) );
+    }
+}
+#endif
--- tools/build/src_ori/engine/jam_strings.h	1970-01-01 00:00:00.000000000 +0000
+++ tools/build/src/engine/jam_strings.h	2026-05-05 16:04:01.339811293 +0000
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2004. David Abrahams
+ * Distributed under the Boost Software License, Version 1.0.
+ * (See accompanying file LICENSE_1_0.txt or copy at
+ * http://www.boost.org/LICENSE_1_0.txt)
+ */
+
+#ifndef STRINGS_DWA20011024_H
+#define STRINGS_DWA20011024_H
+
+#include "config.h"
+#include <stddef.h>
+
+typedef struct string
+{
+    char * value;
+    unsigned long size;
+    unsigned long capacity;
+    char opt[ 32 ];
+#ifndef NDEBUG
+    char magic[ 4 ];
+#endif
+} string;
+
+void string_new( string * );
+void string_copy( string *, char const * );
+void string_free( string * );
+void string_append( string *, char const * );
+void string_append_range( string *, char const *, char const * );
+void string_push_back( string * s, char x );
+void string_reserve( string *, size_t );
+void string_truncate( string *, size_t );
+void string_pop_back( string * );
+char string_back( string * );
+void string_rtrim( string * );
+void string_unit_test();
+
+#endif
--- tools/build/src_ori/engine/modules/order.cpp	2026-05-05 16:01:50.879134398 +0000
+++ tools/build/src/engine/modules/order.cpp	2026-05-05 16:04:01.340811298 +0000
@@ -8,7 +8,7 @@
 #include "../mem.h"
 #include "../native.h"
 #include "../object.h"
-#include "../strings.h"
+#include "../jam_strings.h"
 #include "../variable.h"
 
 
--- tools/build/src_ori/engine/modules/regex.cpp	2026-05-05 16:01:50.879134398 +0000
+++ tools/build/src/engine/modules/regex.cpp	2026-05-05 16:04:01.340811298 +0000
@@ -7,7 +7,7 @@
 
 #include "../mem.h"
 #include "../native.h"
-#include "../strings.h"
+#include "../jam_strings.h"
 #include "../subst.h"
 
 /*
--- tools/build/src_ori/engine/modules.cpp	2026-05-05 16:01:50.878134393 +0000
+++ tools/build/src/engine/modules.cpp	2026-05-05 16:04:01.340811298 +0000
@@ -14,7 +14,7 @@
 #include "object.h"
 #include "parse.h"
 #include "rules.h"
-#include "strings.h"
+#include "jam_strings.h"
 #include "variable.h"
 
 #include <assert.h>
--- tools/build/src_ori/engine/pathsys.h	2026-05-05 16:01:50.879134398 +0000
+++ tools/build/src/engine/pathsys.h	2026-05-05 16:04:01.337811283 +0000
@@ -23,7 +23,7 @@
 
 #include "config.h"
 #include "object.h"
-#include "strings.h"
+#include "jam_strings.h"
 
 
 typedef struct _pathpart
--- tools/build/src_ori/engine/search.cpp	2026-05-05 16:01:50.878134393 +0000
+++ tools/build/src/engine/search.cpp	2026-05-05 16:04:01.342811308 +0000
@@ -20,7 +20,7 @@
 #include "lists.h"
 #include "object.h"
 #include "pathsys.h"
-#include "strings.h"
+#include "jam_strings.h"
 #include "timestamp.h"
 #include "variable.h"
 #include "output.h"
--- tools/build/src_ori/engine/strings.cpp	2026-05-05 16:01:50.880134404 +0000
+++ tools/build/src/engine/strings.cpp	1970-01-01 00:00:00.000000000 +0000
@@ -1,252 +0,0 @@
-/* Copyright David Abrahams 2004. Distributed under the Boost */
-/* Software License, Version 1.0. (See accompanying */
-/* file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt) */
-
-#include "jam.h"
-#include "strings.h"
-
-#include <assert.h>
-#include <stdlib.h>
-#include <string.h>
-
-
-#ifndef NDEBUG
-# define JAM_STRING_MAGIC ((char)0xcf)
-# define JAM_STRING_MAGIC_SIZE 4
-static void assert_invariants( string * self )
-{
-    int i;
-
-    if ( self->value == 0 )
-    {
-        assert( self->size == 0 );
-        assert( self->capacity == 0 );
-        assert( self->opt[ 0 ] == 0 );
-        return;
-    }
-
-    assert( self->size < self->capacity );
-    assert( ( self->capacity <= sizeof( self->opt ) ) == ( self->value == self->opt ) );
-    assert( self->value[ self->size ] == 0 );
-    /* String objects modified manually after construction to contain embedded
-     * '\0' characters are considered structurally valid.
-     */
-    assert( strlen( self->value ) <= self->size );
-
-    for ( i = 0; i < 4; ++i )
-    {
-        assert( self->magic[ i ] == JAM_STRING_MAGIC );
-        assert( self->value[ self->capacity + i ] == JAM_STRING_MAGIC );
-    }
-}
-#else
-# define JAM_STRING_MAGIC_SIZE 0
-# define assert_invariants(x) do {} while (0)
-#endif
-
-
-void string_new( string * s )
-{
-    s->value = s->opt;
-    s->size = 0;
-    s->capacity = sizeof( s->opt );
-    s->opt[ 0 ] = 0;
-#ifndef NDEBUG
-    memset( s->magic, JAM_STRING_MAGIC, sizeof( s->magic ) );
-#endif
-    assert_invariants( s );
-}
-
-
-void string_free( string * s )
-{
-    assert_invariants( s );
-    if ( s->value != s->opt )
-        BJAM_FREE( s->value );
-    string_new( s );
-}
-
-
-static void string_reserve_internal( string * self, size_t capacity )
-{
-    if ( self->value == self->opt )
-    {
-        self->value = (char *)BJAM_MALLOC_ATOMIC( capacity +
-            JAM_STRING_MAGIC_SIZE );
-        self->value[ 0 ] = 0;
-        strncat( self->value, self->opt, sizeof(self->opt) );
-        assert( strlen( self->value ) <= self->capacity && "Regression test" );
-    }
-    else
-    {
-        self->value = (char *)BJAM_REALLOC( self->value, capacity +
-            JAM_STRING_MAGIC_SIZE );
-    }
-#ifndef NDEBUG
-    memcpy( self->value + capacity, self->magic, JAM_STRING_MAGIC_SIZE );
-#endif
-    self->capacity = capacity;
-}
-
-
-void string_reserve( string * self, size_t capacity )
-{
-    assert_invariants( self );
-    if ( capacity <= self->capacity )
-        return;
-    string_reserve_internal( self, capacity );
-    assert_invariants( self );
-}
-
-
-static void extend_full( string * self, char const * start, char const * finish )
-{
-    size_t new_size = self->capacity + ( finish - start );
-    size_t new_capacity = self->capacity;
-    size_t old_size = self->capacity;
-    while ( new_capacity < new_size + 1)
-        new_capacity <<= 1;
-    string_reserve_internal( self, new_capacity );
-    memcpy( self->value + old_size, start, new_size - old_size );
-    self->value[ new_size ] = 0;
-    self->size = new_size;
-}
-
-static void maybe_reserve( string * self, size_t new_size )
-{
-    size_t capacity = self->capacity;
-    if ( capacity <= new_size )
-    {
-        size_t new_capacity = capacity;
-        while ( new_capacity <= new_size )
-            new_capacity <<= 1;
-        string_reserve_internal( self, new_capacity );
-    }
-}
-
-
-void string_append( string * self, char const * rhs )
-{
-    size_t rhs_size = strlen( rhs );
-    size_t new_size = self->size + rhs_size;
-    assert_invariants( self );
-
-    maybe_reserve( self, new_size );
-
-    memcpy( self->value + self->size, rhs, rhs_size + 1 );
-    self->size = new_size;
-
-    assert_invariants( self );
-}
-
-
-void string_append_range( string * self, char const * start, char const * finish )
-{
-    size_t rhs_size = finish - start;
-    size_t new_size = self->size + rhs_size;
-    assert_invariants( self );
-
-    maybe_reserve( self, new_size );
-
-    if ( start != finish )
-        memcpy( self->value + self->size, start, rhs_size );
-    self->size = new_size;
-    self->value[ new_size ] = 0;
-
-    assert_invariants( self );
-}
-
-
-void string_copy( string * s, char const * rhs )
-{
-    string_new( s );
-    string_append( s, rhs );
-}
-
-void string_truncate( string * self, size_t n )
-{
-    assert_invariants( self );
-    assert( n <= self->capacity );
-    self->value[ self->size = n ] = 0;
-    assert_invariants( self );
-}
-
-
-void string_pop_back( string * self )
-{
-    string_truncate( self, self->size - 1 );
-}
-
-
-void string_push_back( string * self, char x )
-{
-    string_append_range( self, &x, &x + 1 );
-}
-
-
-char string_back( string * self )
-{
-    assert_invariants( self );
-    return self->value[ self->size - 1 ];
-}
-
-void string_rtrim( string * self )
-{
-    char *p;
-    assert_invariants( self );
-    p = self->value + self->size - 1;
-    for ( ; p >= self->value && ( *p == '\0' || isspace( *p ) ); *p-- = 0 );
-}
-
-#ifndef NDEBUG
-void string_unit_test()
-{
-    {
-        string s[ 1 ];
-        int i;
-        int const limit = sizeof( s->opt ) * 2 + 2;
-        string_new( s );
-        assert( s->value == s->opt );
-        for ( i = 0; i < limit; ++i )
-        {
-            string_push_back( s, (char)( i + 1 ) );
-            assert( s->size == i + 1 );
-        }
-        assert( s->size == limit );
-        assert( s->value != s->opt );
-        for ( i = 0; i < limit; ++i )
-            assert( s->value[ i ] == (char)( i + 1 ) );
-        string_free( s );
-    }
-
-    {
-        const char * const original = "  \n\t\v  Foo \r\n\v \tBar\n\n\r\r\t\n\v\t \t";
-        string copy[ 1 ];
-        string_copy( copy, original );
-        assert( !strcmp( copy->value, original ) );
-        assert( copy->size == strlen( original ) );
-        string_free( copy );
-    }
-
-    {
-        const char * const foo = "Foo    ";
-        string foo_copy[ 1 ];
-        string_copy( foo_copy, foo );
-        string_rtrim( foo_copy );
-        assert( !strcmp( foo_copy->value, "Foo" ) );
-
-        string_rtrim( foo_copy );
-        assert( !strcmp( foo_copy->value, "Foo" ) );
-    }
-    {
-        const char * const bar = "Bar\0\0\0";
-        string bar_copy[ 1 ];
-        string_copy( bar_copy, bar );
-        string_rtrim( bar_copy );
-        assert( !strcmp( bar_copy->value, "Bar" ) );
-
-        string_rtrim( bar_copy );
-        assert( !strcmp( bar_copy->value, "Bar" ) );
-    }
-}
-#endif
--- tools/build/src_ori/engine/strings.h	2026-05-05 16:01:50.878134393 +0000
+++ tools/build/src/engine/strings.h	1970-01-01 00:00:00.000000000 +0000
@@ -1,38 +0,0 @@
-/*
- * Copyright 2004. David Abrahams
- * Distributed under the Boost Software License, Version 1.0.
- * (See accompanying file LICENSE_1_0.txt or copy at
- * http://www.boost.org/LICENSE_1_0.txt)
- */
-
-#ifndef STRINGS_DWA20011024_H
-#define STRINGS_DWA20011024_H
-
-#include "config.h"
-#include <stddef.h>
-
-typedef struct string
-{
-    char * value;
-    unsigned long size;
-    unsigned long capacity;
-    char opt[ 32 ];
-#ifndef NDEBUG
-    char magic[ 4 ];
-#endif
-} string;
-
-void string_new( string * );
-void string_copy( string *, char const * );
-void string_free( string * );
-void string_append( string *, char const * );
-void string_append_range( string *, char const *, char const * );
-void string_push_back( string * s, char x );
-void string_reserve( string *, size_t );
-void string_truncate( string *, size_t );
-void string_pop_back( string * );
-char string_back( string * );
-void string_rtrim( string * );
-void string_unit_test();
-
-#endif
--- tools/build/src_ori/engine/timestamp.cpp	2026-05-05 16:01:50.880134404 +0000
+++ tools/build/src/engine/timestamp.cpp	2026-05-05 16:04:01.340811298 +0000
@@ -31,7 +31,7 @@
 #include "hash.h"
 #include "object.h"
 #include "pathsys.h"
-#include "strings.h"
+#include "jam_strings.h"
 #include "output.h"
 
 
--- tools/build/src_ori/engine/variable.cpp	2026-05-05 16:01:50.878134393 +0000
+++ tools/build/src/engine/variable.cpp	2026-05-05 16:04:01.341811303 +0000
@@ -38,7 +38,7 @@
 #include "modules.h"
 #include "parse.h"
 #include "pathsys.h"
-#include "strings.h"
+#include "jam_strings.h"
 #include "output.h"
 
 #include <stdio.h>
--- tools/build/src_ori/engine/w32_getreg.cpp	2026-05-05 16:01:50.879134398 +0000
+++ tools/build/src/engine/w32_getreg.cpp	2026-05-05 16:04:01.344811319 +0000
@@ -12,7 +12,7 @@
 # include "object.h"
 # include "parse.h"
 # include "frames.h"
-# include "strings.h"
+# include "jam_strings.h"
 
 # define WIN32_LEAN_AND_MEAN
 # include <windows.h>

--- tools/build/src_ori/engine/build.bat	2026-05-05 16:01:50.877134388 +0000
+++ tools/build/src/engine/build.bat	2026-05-05 16:34:11.087110346 +0000
@@ -174,7 +174,7 @@
 set B2_SOURCES=%B2_SOURCES% glob.cpp hash.cpp hcache.cpp hdrmacro.cpp headers.cpp jam.cpp
 set B2_SOURCES=%B2_SOURCES% jambase.cpp jamgram.cpp lists.cpp make.cpp make1.cpp md5.cpp mem.cpp modules.cpp
 set B2_SOURCES=%B2_SOURCES% native.cpp object.cpp option.cpp output.cpp parse.cpp pathnt.cpp
-set B2_SOURCES=%B2_SOURCES% pathsys.cpp regexp.cpp rules.cpp scan.cpp search.cpp strings.cpp
+set B2_SOURCES=%B2_SOURCES% pathsys.cpp regexp.cpp rules.cpp scan.cpp search.cpp jam_strings.cpp
 set B2_SOURCES=%B2_SOURCES% subst.cpp sysinfo.cpp timestamp.cpp variable.cpp w32_getreg.cpp
 set B2_SOURCES=%B2_SOURCES% modules/order.cpp
 set B2_SOURCES=%B2_SOURCES% modules/path.cpp
--- tools/build/src_ori/engine/build.sh	2026-05-05 16:01:50.877134388 +0000
+++ tools/build/src/engine/build.sh	2026-05-05 16:33:55.927031445 +0000
@@ -413,7 +413,7 @@
  rules.cpp \
  scan.cpp \
  search.cpp \
- strings.cpp \
+ jam_strings.cpp \
  subst.cpp \
  sysinfo.cpp \
  timestamp.cpp \
