diff -Naur CGNS-4.4.0/CMakeLists.txt CGNS-4.4.0-new/CMakeLists.txt
--- CGNS-4.4.0/CMakeLists.txt	2023-06-15 05:00:19.000000000 +0100
+++ CGNS-4.4.0-new/CMakeLists.txt	2024-10-21 11:21:02.742057896 +0100
@@ -251,7 +251,8 @@
   message (STATUS "HDF5 find comps: ${FIND_HDF_COMPONENTS}")
 
   set (SEARCH_PACKAGE_NAME "hdf5")
-  
+  set (HDF5_PREFER_PARALLEL ON)
+
   # If the first `find_package` below does not succeed, then the legacy `find_package`
   # is tried (the `else` below).  The legacy find_package uses `HDF5_ROOT`.  But if 
   # this is set, then CMake will issue warning and mistakenly say that `HDF5_ROOT` is
