--- gsmpi-1.2.1_2/CMakeLists.txt	2026-06-04 09:45:44.930366200 +0100
+++ gsmpi-1.2.1_2/CMakeLists.txt	2026-06-04 09:45:44.930366200 +0100
@@ -1,4 +1,4 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.7)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.5)
 
 # compile in Release mode by default
 SET(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None(CMAKE_CXX_FLAGS or CMAKE_C_FLAGS used) Debug Release RelWithDebInfo MinSizeRel.")
