--- lapack-3.7.1/CMakeLists.txt	2026-06-04 11:03:57.518890772 +0100
+++ lapack-3.7.1/CMakeLists.txt	2026-06-04 11:03:57.518890772 +0100
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.10)
+cmake_minimum_required(VERSION 3.5)
 
 # Set a default build type if none was specified
 if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
--- lapack-3.7.1/INSTALL/CMakeLists.txt	2026-06-04 16:13:55.522018700 +0100
+++ lapack-3.7.1/INSTALL/CMakeLists.txt	2026-06-04 16:13:55.522018700 +0100
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.7)
+cmake_minimum_required(VERSION 3.5)
 project(TIMING Fortran)
 add_executable(secondtst_NONE second_NONE.f secondtst.f)
 add_executable(secondtst_EXT_ETIME second_EXT_ETIME.f secondtst.f)
